modules/apps/27-interchain-accounts/controller/keeper/handshake.go

```diff
@@ -72,10 +72,14 @@ func (k Keeper) OnChanOpenInit(
 			panic(fmt.Errorf("active channel mapping set for %s but channel does not exist in channel store", activeChannelID))
 		}
 
-		if channel.IsOpen() {
+		if channel.State == channeltypes.OPEN {
```

Comment:
Thinking now if we should change this condition to `channel.State != channeltypes.CLOSED`... I am thinking of the following scenario: an ICA channel is going through a channel upgrade and the channel state is either `FLUSHING` or `FLUSH_COMPLETE`, and a relayer submits a `MsgChannelOpenInit` with the same metadata version as the channel going through the upgrade. With the current check the opening handshake would go through, right? I guess we should prevent that?

If we need to change this, we can do it in a follow up PR.

Reply:
good call, maybe we can create a follow up issues for it, (this PR is not changing any logic in this condition)

Reply:
This condition looks like it's already being handled [here](https://github.com/cosmos/ibc-go/blob/d58cf6f057b4df8986f7328622b6033995ca8a46/modules/apps/27-interchain-accounts/controller/keeper/handshake.go#L88-L90)

Reply:
wouldn't it fail since channel open would try and claim and already claimed port cap?

Reply:
`ChanOpenInit` on the controller doesn't seem to claim any capabilities, this happens in `OnChanOpenTry` on the host chain

Reply:
this seems like a valid concern to me. We should change to != CLOSED. Active channel gets set upon channel opening so we don't need to worry about edge cases with INIT and TRY

Reply:
> wouldn't it fail since channel open init would try and claim and already claimed port cap?

The capability is claimed [here](https://github.com/cosmos/ibc-go/blob/main/modules/apps/27-interchain-accounts/controller/ibc_middleware.go#L68), but the thing is that when reopening a closed ICA channel we are actually creating a new with the same metadata information as the previous one, so that's why claiming the capability will not fail.

> wouldn't it fail since channel open init would try and claim and already claimed port cap?

I will open an issue.